test/ui/package.json

```diff
@@ -6,6 +6,7 @@
     "run-e2e": "$(cd ../.. && scripts/run-local.sh ui &) && cypress run"
   },
   "devDependencies": {
-    "cypress": "^13.6.6"
+    "cypress": "^13.6.6",
+    "typescript": "^5.4.2"
```

Comment:
We should not use typescript. Its just another dependency and code needs to be compiled from ts to js. Looking at the code we use it so little its not worth it.

Reply:
@tznamena, it is better to use typescript, which enhances code quality and maintainability and improve productivity. :-) We already have the typescript used in this repo for the ui codes (<repo>/ui).

Reply:
We are not using typescript at all in the code, let's not add complexity for no reason.

Reply:
@tznamena , before I started to use typescript, I took a look at <repo>/ui, where it has the typescript installed. https://github.com/ansible/azure-aap-deployment-driver/blob/main/ui/package-lock.json#L29

Reply:
That's a different project, with much bigger code base and much higher complexity. We do not need typescript here.

Reply:
@tznamena , there are two reasons that I want to keep typescript:
1) It doesn't increase our codes' complexity. Instead it enhances code quality and maintainability. It can improve our coding productivity by referring what it suggests when we are typing the codes.
2) The Platform UI QE automation uses typescript. Using typescript to automate deployment engine ui will be easier for QE to maintain all front end Cypress automation test suites.

Reply:
This is a open source project and I'm trying to keep the code base clean. The approach of using something just because some other team uses it somewhere else is simply wrong.
Let's remove typescript (this PR uses TS in two places only!!!) and we can add it back when it becomes necessary.

Reply:
@ryankwilliams @bhavenst , what's your thoughts about the typescript?

Reply:
Just for everyone's awareness:
1. There is NO typescript used in any of the test files! Not a single keyword!
2. There is one piece of typescript in `test/ui/cypress/support/auth.ts` and that is: `requiredVariables: string[]`
3. There is one file that's written in typescript, `test/ui/cypress/support/commands.ts` however that file does not add any functionality.

Again, this is an open source project and we will not be adding dependencies that do not add value or just because they are used elsewhere. When the need to use Typescript comes up, we can add it. At this point, it is completely unnecessary.

Reply:
@tznamena , what do you mean that there is no typescript used in any of the test files? If you look at the cypress spec test files that I created, all of them have a `.ts` suffix in their file names, which enable the typescript features to them.

Reply:
Naming a file with `.ts` only indicates that file might contain typescript, however, all the code in all the test files is 100% Javascript. And combine that fact with the fact that browsers only run Javascript, it must be pretty obvious to everyone that we do NOT need typescript in this part of the code base.

Reply:
@tznamena , I tried to remove the typescript dependency from my codes and started to add the command commands in the `commands.js' instead of using `commands.ts`, my automation started to fail to run. If we really want to get the typescript dependency removed, can we get this PR approved & merged then I will submit another story to work on it?

Reply:
You can just remove the `test/ui/cypress/support/commands.ts` file, and add following to the `test/ui/cypress/support/e2e.js` file:
```js
import './auth'
import './logout'
```

Reply:
Unfortunately it didn't work. This is the error that I got after I made change as what you suggested above.
`Error: Webpack Compilation Error
Module build failed (from ../../../../../.cache/Cypress/13.6.6/Cypress/resources/app/node_modules/babel-loader/lib/index.js):
SyntaxError: /home/qwa/projects/qiwangrh/azure-aap-deployment-driver/test/ui/cypress/support/auth.js: Unexpected token, expected "," (1:66)

> 1 | Cypress.Commands.add('requiredVariablesAreSet', (requiredVariables: string[]) => {
    |                                                                   ^
  2 |     requiredVariables.forEach((variable) => {
  3 |       if (!Cypress.env(variable)) {
  4 |         throw new Error(`Missing required environment variable: ${variable}`);
    at Watching.handle [as handler] (/home/qwa/.cache/Cypress/13.6.6/Cypress/resources/app/node_modules/@packages/server/node_modules/@cypress/webpack-preprocessor/dist/index.js:212:23)
    at /home/qwa/.cache/Cypress/13.6.6/Cypress/resources/app/node_modules/webpack/lib/Watching.js:303:9
    at Hook.eval [as callAsync] (eval at create (/home/qwa/.cache/Cypress/13.6.6/Cypress/resources/app/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
    at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (/home/qwa/.cache/Cypress/13.6.6/Cypress/resources/app/node_modules/tapable/lib/Hook.js:18:14)
    at Watching._done (/home/qwa/.cache/Cypress/13.6.6/Cypress/resources/app/node_modules/webpack/lib/Watching.js:299:28)
    at /home/qwa/.cache/Cypress/13.6.6/Cypress/resources/app/node_modules/webpack/lib/Watching.js:213:21
    at Compiler.emitRecords (/home/qwa/.cache/Cypress/13.6.6/Cypress/resources/app/node_modules/webpack/lib/Compiler.js:919:5)
    at /home/qwa/.cache/Cypress/13.6.6/Cypress/resources/app/node_modules/webpack/lib/Watching.js:191:22
    at /home/qwa/.cache/Cypress/13.6.6/Cypress/resources/app/node_modules/webpack/lib/Compiler.js:885:14
    at Hook.eval [as callAsync] (eval at create (/home/qwa/.cache/Cypress/13.6.6/Cypress/resources/app/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
    at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (/home/qwa/.cache/Cypress/13.6.6/Cypress/resources/app/node_modules/tapable/lib/Hook.js:18:14)
    at /home/qwa/.cache/Cypress/13.6.6/Cypress/resources/app/node_modules/webpack/lib/Compiler.js:882:27
    at /home/qwa/.cache/Cypress/13.6.6/Cypress/resources/app/node_modules/neo-async/async.js:2818:7
    at done (/home/qwa/.cache/Cypress/13.6.6/Cypress/resources/app/node_modules/neo-async/async.js:3522:9)
    at Hook.eval [as callAsync] (eval at create (/home/qwa/.cache/Cypress/13.6.6/Cypress/resources/app/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
    at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (/home/qwa/.cache/Cypress/13.6.6/Cypress/resources/app/node_modules/tapable/lib/Hook.js:18:14)
    at /home/qwa/.cache/Cypress/13.6.6/Cypress/resources/app/node_modules/webpack/lib/Compiler.js:736:33
    at /home/qwa/.cache/Cypress/13.6.6/Cypress/resources/app/node_modules/graceful-fs/graceful-fs.js:143:16
    at /home/qwa/.cache/Cypress/13.6.6/Cypress/resources/app/node_modules/@packages/server/node_modules/graceful-fs/graceful-fs.js:143:16
    at /home/qwa/.cache/Cypress/13.6.6/Cypress/resources/app/node_modules/@packages/server/node_modules/graceful-fs/graceful-fs.js:143:16
    at /home/qwa/.cache/Cypress/13.6.6/Cypress/resources/app/node_modules/@packages/server/node_modules/graceful-fs/graceful-fs.js:61:14
    at FSReqCallback.oncomplete (node:fs:189:23)`

Reply:
@tznamena , the custom command `requiredVariablesAreSet` reported the error since Type annotations can only be used in typescript. If we really want to remove the typescript from the deployment engine UI, I might need a new story to troubleshot the failure and retest the codes to get the automated test scenarios working again without typescript.
`Cypress.Commands.add('requiredVariablesAreSet', (requiredVariables: string[]) => {
    requiredVariables.forEach((variable) => {
      if (!Cypress.env(variable)) {
        throw new Error(`Missing required environment variable: ${variable}`);
      }
    });
});`

Reply:
Removed the custom command.